### PR TITLE
Use Docker-Meta for compliance

### PR DESCRIPTION
**Pull request by jmcs**
As part of handling Zalando specific details in an internal service:

- Get as much custom information from docker-meta as possible
- Mark production ready with pierone.

### CODE REVIEW

File: pierone/api.py [hunk 1]
```diff
@@ -17,95 +18,136 @@
...
+    ):
+        path = "/image-metadata/{}/{}/{}:{}".format(
+            image.registry, image.team, image.artifact, image.tag
+        )
+        payload = {
+            "compliance": {
+                "user": {
+                    "incident": incident_id,
+                    "reason": reason,
+                    "status": "production-ready",
```
Comment (omartrigui) [inline pierone/api.py:112]:
```suggestion
                    "status": "production_ready",
```


```python
class Status(str, Enum):
    production_ready = "production_ready"
    stups_production_ready = "stups_production_ready"
    not_production_ready = "not_production_ready"
```

File: pierone/api.py [hunk 2]
```diff
@@ -17,95 +18,136 @@
...
+                }
+            }
+        }
+        self._put(path, json=payload)
+
+    def get_image_metadata(self, image: DockerImage) -> dict:
+        """
+        Gets all tne image's metadata that Pierone-CLI uses.
```
Comment (omartrigui) [inline pierone/api.py:120]:
```suggestion
        Gets all the image's metadata that Pierone-CLI uses.
```

File: pierone/cli.py [hunk 3]
```diff
@@ -227,19 +227,26 @@ def mark_production_ready(config, incident, team, artifact, tag, url):
     pierone_url = set_pierone_url(config, url)
     registry = get_registry(pierone_url)
     image = DockerImage(registry, team, artifact, tag)
-    api = PierOne(pierone_url)
-    api.mark_production_ready(image, incident)
+    if incident.startswith("INC-"):
+        # if it's a JIRA ticket mark image as production ready in pierone
```
Comment (omartrigui) [inline pierone/cli.py:231]:
```suggestion
        # if it's a JIRA ticket, mark image as production ready in pierone
```

File: pierone/api.py [hunk 4]
```diff
@@ -17,95 +18,136 @@
...
-        except requests.HTTPError as error:
-            self._handle_exceptions(error, exceptions)
-        return response
+        return self._request("GET", path, *args, exceptions=exceptions, **kwargs)
+
+    def _put(
+        self, path, json=None, exceptions: Optional[dict] = None, *args, **kwargs
+    ) -> requests.Response:
+        """
+        PUTs things to Pier One and Docker Meta.
```
Comment (omartrigui) [inline pierone/api.py:72]:
```suggestion
        PUTs things to Pierone and Docker Meta.
```

**Comment by jmcs** [comment]
:+1:

File: pierone/cli.py [hunk 5]
```diff
@@ -227,19 +227,26 @@ def mark_production_ready(config, incident, team, artifact, tag, url):
     pierone_url = set_pierone_url(config, url)
     registry = get_registry(pierone_url)
     image = DockerImage(registry, team, artifact, tag)
-    api = PierOne(pierone_url)
-    api.mark_production_ready(image, incident)
+    if incident.startswith("INC-"):
+        # if it's a JIRA ticket, mark image as production ready in pierone
```
Comment (omartrigui) [inline pierone/cli.py:231]:
```suggestion
        # if it's a JIRA ticket, mark image as production ready in Pierone
```

File: pierone/api.py [hunk 6]
```diff
@@ -180,82 +247,100 @@ def load_docker_config():
...
-def request(url, path, access_token: str = None,
-            not_found_is_none: bool = False, method: str = 'GET', data=None) -> requests.Response:
+def request(
+    url,
+    path,
+    access_token: str = None,
+    not_found_is_none: bool = False,
+    method: str = "GET",
+    data=None,
+) -> requests.Response:
```
Comment (omartrigui) [inline pierone/api.py:335]:
```suggestion
) -> Optional[requests.Response]:
```
We have [return None](https://github.com/zalando-stups/pierone-cli/pull/97/files#diff-54b230de210663e741305792c3396e44ab9165a0fd1819e9bf5057bb1a2543dfR346)

**Comment by omartrigui** [comment]
:+1:

**Comment by jmcs** [comment]
:+1:

**Pull request merged**